introspection-engine/introspection-engine-tests/tests/multi_schema/postgres.rs

```diff
@@ -0,0 +1,421 @@
+        }
+
+        model HappyPerson {
+          mood HappyMood @id
+
+          @@schema("first")
+        }
+
+        model HappyPerson {
+          mood HappyMood @id
+
+          @@schema("second")
+        }
+
+        model VeryHappyPerson {
+          mood HappyMood @id
+
+          @@schema("second")
+        }
+
+        enum HappyMood {
+          happy
+
+          @@schema("first")
+        }
+
+        enum HappyMood {
+          veryHappy
+
+          @@schema("second")
+        }
+    "#]];
+
+    api.expect_datamodel(&expected).await;
+    Ok(())
+}
+
+#[test_connector(tags(Postgres))]
+async fn multiple_schemas_w_enums_without_schemas_are_not_introspected(api: &TestApi) -> TestResult {
+    let schema_name = api.schema_name();
+    let other_name = "second";
+    let create_type = format!("CREATE TYPE \"{schema_name}\".\"HappyMood\" AS ENUM ('happy')",);
+
+    api.database().raw_cmd(&create_type).await?;
+
+    let create_schema = format!("CREATE Schema \"{other_name}\"",);
+    let create_type = format!("CREATE TYPE \"{other_name}\".\"SadMood\" AS ENUM ('sad')",);
+
+    api.database().raw_cmd(&create_schema).await?;
+    api.database().raw_cmd(&create_type).await?;
+
+    let expected = expect![[r#"
+        enum HappyMood {
+          happy
+        }
+    "#]];
+
+    let result = api.introspect_dml().await?;
+    expected.assert_eq(&result);
+
+    Ok(())
+}
+
+#[test_connector(
+    tags(Postgres),
+    exclude(CockroachDb),
+    preview_features("multiSchema"),
+    namespaces("first", "second_schema")
+)]
+async fn same_table_name_with_relation_in_two_schemas(api: &TestApi) -> TestResult {
+    let sql = r#"
+        CREATE SCHEMA "first";
+        CREATE SCHEMA "second_schema";
+        CREATE TABLE "first"."tbl" ( id SERIAL PRIMARY KEY );
+        CREATE TABLE "second_schema"."tbl" ( id SERIAL PRIMARY KEY, fst INT REFERENCES "first"."tbl"("id") );
+    "#;
+
+    api.raw_cmd(&sql).await;
+
+    let expected = expect![[r#"
+        generator client {
+          provider        = "prisma-client-js"
+          previewFeatures = ["multiSchema"]
+        }
+
+        datasource db {
+          provider = "postgresql"
+          url      = "env(TEST_DATABASE_URL)"
+          schemas  = ["first", "second_schema"]
+        }
+
+        model tbl {
+          id  Int   @id @default(autoincrement())
+          tbl tbl[]
+
+          @@schema("first")
+        }
+
+        model tbl {
+          id  Int  @id @default(autoincrement())
+          fst Int?
+          tbl tbl? @relation(fields: [fst], references: [id], onDelete: NoAction, onUpdate: NoAction)
+
+          @@schema("second_schema")
+        }
+    "#]];
+
+    api.expect_datamodel(&expected).await;
+    Ok(())
+}
+
+//cross schema
+// fks
+// enums
+
+//Edge cases
+//name conflicts
+// what if the names are used somewhere???
+// table
+// enum
+//invalid names
+// schema
+// table
+// enum
+// re-introspection
+// table
+// enum
```

Comment:
Should this be here? If yes, could you elaborate such that it makes sense for everyone?

Or maybe it needs to be a GH issue or something?

Reply:
It was just copied from the other file. I do not know should it be here, but also didn't want to delete it.